### DISCUSSION TITLE
Interface problem after security_force

### DISCUSSION THREAD

**Discussion by MurzzZ** [rgerganov/gen5fw#18] [Q&A]
Hi all!
First of all, thank you so much for a great job!
I installed security_force on my device, but got unexpected changes in the interface, see photo.
ADB is available after 5 clicks, there are no problems with this.
The interface is restored by a standard update, but after that ADB does not work.
Is there any way to patch the update so that ADB remains available?

I must say that I made a rather unusual update. Initially, my device is for Kia sportage 18 USA, and there are no maps of Europe in it. Everything that is in Europe for Kia works on LG devices. But European Hyundai works with mobis, so I tried to install mobis firmware from European tucson and it works!
![mobis](https://github.com/rgerganov/gen5fw/assets/31242701/0ce854cd-4263-4077-b865-805d687cea71)
![mobis1](https://github.com/rgerganov/gen5fw/assets/31242701/d4b6d5a6-fc58-4584-897d-2c8c13d09536)

**Comment by al3emsy** [comment]
Wait so you managed to flash firmware intended for Hyundai Tucson into a Kia Sportage? after flashing the new firmware you installed the european maps using the original maps and firmware through SD which is provided by Hyundai navigation updater?

you mention that the interface is restored after a standard update, I am guessing you mean by updating through the official navigation updater, and after that step the ADB is disabled once again? am I getting this correctly?

I installed security_force on my device, but got unexpected changes in the interface, see photo.
by this you mean the firmware version, etc? what is unexpected?

**Comment by MurzzZ** [comment]
That's right, I compared the hardware versions on the American Sportage and the European Tucson and came to the conclusion that this is the same hardware platform, with minor changes. For example, in the US there is no wifi.
After that, I downloaded the original update for Hyundai tucson, made a security_force and updated my american Sportage unit.
But as you can see on the photo in first message, in the GUI many items have the same name "Manage applications", but ADB is available.
I updated the with the original Tucson update again, GUI and maps of Europe were completely restored as in the original Hyundai, but the adb was gone.
![2 — копия](https://github.com/rgerganov/gen5fw/assets/31242701/58b6c63e-6b00-4321-87d1-3dc43f8abe4d)
![3 — копия](https://github.com/rgerganov/gen5fw/assets/31242701/9aa88fee-0046-4649-a075-d08fda7c918b)
![4 — копия](https://github.com/rgerganov/gen5fw/assets/31242701/6a670d7f-f4af-42cf-88bc-fad3b2b069b2)
![1 — копия](https://github.com/rgerganov/gen5fw/assets/31242701/0c7ad9b0-f54f-4191-8fc8-ffdfa50bbf48)

**Comment by al3emsy** [comment]
This is the first progress I have seen so far about this topic, I have seen people report having all names changing to manage applications, but you are the first I saw to give a report that updating with OEM firmware reverts everything back to normal, its more like using the force_security to gain backdoor access and allow the unit to accept other region firmware, as for losing privilege's after update, especially ADB control in your situation, I think its related to the OEM firmware over-writing and reverting all settings back to original state.

I have been trying to do this process on my car and have been a bit skeptical about it, is there a way or a platform where I could message you on, I have a couple of questions regarding process and steps if you wouldn't mind!

**Comment by MurzzZ** [comment]
You can write me in telegram https://t.me/alexey_murzin

**Comment by rgerganov** [comment]
> The interface is restored by a standard update, but after that ADB does not work.
> Is there any way to patch the update so that ADB remains available?

You can try another security_force update with a patched European firmware. I believe the issue with "Manage applications" appears only when forcing firmware from another region.

If you decide to go with another security_force update, please let us know how it went. Thanks.

**Comment by MurzzZ** [comment]
I've tried secutity_update a couple of times with the current Tucson firmware installed, but always the same result.
By the way, we have a hw UART on unit, you can read about it on the Russian 4da forum: https://4pda.to/forum/index.php?showtopic=951764&view=findpost&p=116731173

**Reply by adelie93** [reply]
I wish I had known about the UART before I did this, I would have taken a full disk image. That would have been very useful in troubleshooting!

Has anyone else taken a full disk image before applying the `security force` update? I would be interested to see what the original partitions looked like.

**Reply by al3emsy** [reply]
In theory wouldn't flashing the original firmware back in then updating through the original update restore original partitions then you can do the UART?

**Reply by adelie93** [reply]
Not in my experience. Something has permanently changed after the `security force` update. I think the normal update process writes the data to the appropriate partitions without moving or re-sizing them, while the force update re-partitions the disk according to `partition.dat`, which may be different than the original. There does not seem to be any partition layout included in the normal update.

**Reply by al3emsy** [reply]
one way you could try and rule out is performing the force_update and running the UART, then update through original firmware and run UART again, then compare two partitions, because after updating whatever modifications you made to system files reverts back to normal, of course I could be wrong, but its a theory!

**Comment by mark0Dex** [comment]
I have the same problem. After installing the EUR system on the EUR device, i have unavailable icons, as if the system was not fully loaded even though i had no errors during installation. Maybe the problem is converting ext4 to iso? What software version are you using?

I'm wondering about this value: (...) lk.rom (...) sector [0]

![20240331_114247](https://github.com/rgerganov/gen5fw/assets/165553717/7ef545c0-a468-4a8d-9cba-fe5a1f26c37b)
![20240331_114553](https://github.com/rgerganov/gen5fw/assets/165553717/3870adec-ed86-4869-addb-33e29a2f22fe)
![20240331_114602](https://github.com/rgerganov/gen5fw/assets/165553717/feb73992-e062-46dc-ab17-2d52e88ac283)
